internal/group/group.service.go

```diff
@@ -33,58 +37,384 @@ func (s *serviceImpl) FindOne(ctx context.Context, in *proto.FindOneGroupRequest
 	cacheKey := fmt.Sprintf("group:%s", in.UserId)
 	var cachedGroup proto.Group
 
-	// try to retreive from cache
+	// Try to retrieve from cache
 	err := s.cache.GetValue(cacheKey, &cachedGroup)
 	if err == nil {
 		s.log.Info("Group found in cache", zap.String("user_id", in.UserId))
 		return &proto.FindOneGroupResponse{Group: &cachedGroup}, nil
 	}
 
+	// If not found in cache, fetch from database
 	userUUID, err := uuid.Parse(in.UserId)
 	if err != nil {
-		return nil, fmt.Errorf("invalid UUID format: %v", err)
+		s.log.Error("Parse: Invalid UUID format", zap.String("user_id", in.UserId), zap.Error(err))
+		return nil, fmt.Errorf("parse invalid UUID format: %w", err)
 	}
 
-	// if not found cache, find group in database
 	group, err := s.repo.FindOne(userUUID)
 	if err != nil {
 		s.log.Error("Failed to find group", zap.String("user_id", in.UserId), zap.Error(err))
-		return nil, err
+		return nil, fmt.Errorf("failed to find group: %w", err)
 	}
 
-	userInfo := make([]*proto.UserInfo, 0, len(group.Members))
-	for _, m := range group.Members {
-		user := proto.UserInfo{
-			Id:        m.ID.String(),
-			Firstname: m.Firstname,
-			Lastname:  m.Lastname,
-			ImageUrl:  m.PhotoUrl,
-		}
-		userInfo = append(userInfo, &user)
+	// Convert to RPC format and update cache
+	groupRPC := s.convertToGroupRPC(group)
+	if err := s.cache.SetValue(cacheKey, groupRPC, 3600); err != nil {
+		s.log.Warn("Failed to set group in cache", zap.String("user_id", in.UserId), zap.Error(err))
```

Comment:
`s.log.Named("FindOne").Error("SetValue: Failed to set group in cache", zap.String("user_id", in.UserId), zap.Error(err))`
please make all the logs in service look smt like this:
`s.log.Name("service_method").Error("function_inside_service_method: blah", zap.Error(err) or other error formats)

Reply:
let me edit come code

Reply:
thank you kub